docs/how-to/uc-image-creation.rst

```diff
@@ -0,0 +1,220 @@
+How to create a real-time Ubuntu Core image
+===========================================
+
+.. note::
+
+    This guide assumes access to features available exclusively to `dedicated Snap Store`_ users.
+
+The :doc:`../how-to/uc-boot-parameters` describes the method for dynamically configuring the system for real-time processing.
+The configuration is an iterative process that is best done together with the expected workload. 
+Once satisfied with the configurations, it is time to prepare the operating system for going into production.
+
+This guide shows how to statically set the desired Kernel command-line options for the Ubuntu Core system.
+To do this, we need to create a custom gadget snap, create a model assertion, and then build the OS image.
+
+.. admonition:: Project directory
+
+    Start in an empty directory.
+    We refer to this in different parts of the document as our *project directory*.
+
+Create the gadget snap
+----------------------
+
+The `gadget snap`_ documentation is a recommended read before starting.
+
+This is best done by forking an existing reference gadget, then changing it for our purpose.
+For example, there is the `pc gadget`_ which is suitable for most AMD64 platforms, and the `pi gadget`_ which is meant for Raspberry Pis.
+
+Let's create a custom gadget snap based on the core22 pc gadget.
+
+Inside the project directory, clone the specific branch and enter the repository:
+
+.. code-block:: shell
+
+    git clone https://github.com/snapcore/pc-gadget.git --branch=22 --depth=1
+    cd pc-gadget
+
+
+Add the desired kernel command line in an array to ``kernel-cmdline.append`` in ``gadget/gadget-amd64.yaml``.
+For example:
+
+.. code-block:: yaml
+
+    kernel-cmdline:
+        append:
+            - nohz=on
+            - nohz_full=2-N
+            - irqaffinity=0-1
+
+
+Refer to :doc:`../reference/kernel-boot-parameters` for the list of supported parameters.
+
+Modify ``snapcraft.yaml`` to fit your application.
+At least, make sure to change the name and version to something distinct, for example, to ``realtime-pc`` and ``example`` respectively.
+
+
+Now, build the gadget snap:
+
+.. code-block:: console
+
+    $ snapcraft --verbose
+    ...
+    Created snap package realtime-pc_example_amd64.snap
+
+
+.. tip::
+    You need to rebuild the snap every time you change the `snapcraft.yaml` file.
+
+
+Create the model assertion
+--------------------------
+
+The model assertion is a digitally signed document that describes the content of the Ubuntu Core image.
+Read the `model assertion`_ documentation before continuing.
+
+Below is an example model assertion in YAML, describing a `core22` Ubuntu Core
+image:
+
+.. literalinclude :: uc-image-creation/model.json
+   :language: json
+
+Create the model assertion inside the project directory.
+
+Set the following:
+
+
+- ``authority-id``, ``brand-id`` to your developer ID
+- ``timestamp`` to a RFC3339 formatted string, within you signing key's validity
```

Comment:
The timestamp need a value post creation date of the snap key right?

Reply:
I believe the timestamp should be within the signing key's validity. The key is a PGP key-pair which can be generated with `snap create-key`, `snapcraft create-key`, or another program. PGP keys can have creation and expiry dates but the one `snap create-key` generates doesn't have the right date, so I'm not sure what date is enforced, maybe it has to do with the registration date into the snap store. Need to check.

Reply:
Details of the public key generated for this how-to:
```
gpg  --homedir /home/farshid/.snap/gnupg/ --export -a realtime-ubuntu | gpg --list-packets
# off=0 ctb=99 tag=6 hlen=3 plen=525
:public key packet:
        version 4, algo 1, created 1451606400, expires 0
        pkey[0]: [4096 bits]
        pkey[1]: [17 bits]
        keyid: A8C4F62D7912B49E
# off=528 ctb=b4 tag=13 hlen=2 plen=15
:user ID packet: "realtime-ubuntu"
# off=545 ctb=89 tag=2 hlen=3 plen=575
:signature packet: algo 1, keyid A8C4F62D7912B49E
        version 4, created 1451606400, md5len 0, sigclass 0x13
        digest algo 10, begin of digest 61 51
        hashed subpkt 33 len 21 (issuer fpr v4 5D2F0F60249592D1A49CB80BA8C4F62D7912B49E)
        hashed subpkt 2 len 4 (sig created 2016-01-01)
        hashed subpkt 27 len 1 (key flags: 2F)
        hashed subpkt 21 len 1 (pref-hash-algos: 10)
        hashed subpkt 30 len 1 (features: 01)
        hashed subpkt 23 len 1 (keyserver preferences: 80)
        subpkt 16 len 8 (issuer key ID A8C4F62D7912B49E)
        data: [4096 bits]
```

Reply:
The timestamp must be after the registration date of the key.